napari/components/add_layers_mixin.py

```diff
@@ -45,6 +50,57 @@ def add_layer(self, layer):
         if len(self.layers) == 1:
             self.reset_view()
 
+    def add_path(
+        self,
+        path: Union[str, Sequence[str]],
+        stack: bool = False,
+        use_dask: Optional[bool] = None,
+    ) -> List[layers.Layer]:
+        """Add a path to the viewer
+
+        Paths will be handed one-by-one to the napari_get_reader hook.
+
+        Parameters
+        ----------
+        path : str or list of str
+            path(s) to view.
+        stack : bool, optional
+            Concatenate multiple input files into a single stack,
+            by default False
+        use_dask : bool, optional
+            Whether to use dask to create a lazy array, rather than NumPy.
+            If ``None``, will resolve to True if filenames contains more than
+            one image, False otherwise.  by default None.
+        """
+        paths = [path] if isinstance(path, str) else path
+        if not isinstance(paths, (tuple, list)):
+            raise ValueError(
+                "'path' argument must be a string, list, or tuple"
+            )
+
+        added_layers: List[layers.Layer] = []
+        if stack:
+            images = io.magic_imread(paths, use_dask=use_dask, stack=stack)
+            added = self.add_image(images)
+            added = added if isinstance(added, list) else [added]
+            added_layers.extend(added)
+            return added
```

Comment:
I can't say I love this, I guess the main reason this can't be a plugin is that plugins can only accept paths instead of also accepting lists of paths. If we allowed lists of paths then this could be accomplished, but maybe that increases complexity of the plugins for something that isn't too bad here, so maybe this is the right choice

Reply:
I'm totally with you @sofroniewn. This seems almost worth removing. @tlambert03 out of the options you presented, I'm definitely leaning towards making our own implementation a plugin. This will help us figure out how plugins are limited.

Reply:
I agree. I'd much prefer to let plugins have access to everything, and this prevents plugins from reading stacks of individual files.  I like @jni's suggestion of having `napari_get_reader` accept either a `str` or `Sequence[str]` and return the appropriate reader function accordingly.  I'll work on that.